teambot.py

```diff
@@ -7,17 +7,21 @@
 import atexit
 
 # This directory maps slack channel IDs (not names) to sets of user IDs
-directory = shelve.open('teams.db')
+directory = None
 
 outputs = []
 my_user_id = None
 slack_client = None
 
-def setup(bot):
+def setup(bot, config):
     global my_user_id
     global slack_client
-    slack_client = bot.slack_client
+    global directory
+
+    dir_file = config.get('TEAM_DB_FILE') or 'teams.db'
```

Comment:
The docs say:
> . As a side-effect, an extension _may_ be added to the filename and more than one file _may_ be created.

This is a WTF for an API if you ask me, it should be predictable what filename your database library is going to use when that's the only way of identifying the database.